src/spatialdata/_core/operations/aggregate.py

```diff
@@ -265,7 +267,8 @@ def _aggregate_shapes(
     ).tocsr()
     return ad.AnnData(
         X,
-        obs=pd.DataFrame(index=obs_id_categorical.categories),
+        # obs=pd.DataFrame(index=obs_id_categorical.categories),
+        obs=pd.DataFrame(index=pd.Categorical(by.index.tolist()).categories),
```

Comment:
why changing this line, is modified in 259 already, would use that, logic is not clear